src/supervisor/metadata-extractor.ts

```diff
@@ -54,6 +59,7 @@ export function buildImageMetadata(
       specAnnotations: specMeta.annotations || {},
       containerName: containerStatus.name,
       imageName: containerNameToSpec[containerStatus.name].image,
+      //! why do we use containerNameToSpec rather than containerNameToStatus here?
```

Comment:
The spec contains the unprocessed data, as the user has requested it when applying the deployment. The status contains the processed data about the running pod, hence imageID shows there (which is the resolved data after pulling the image).

I had a look at a few examples and in both the spec and in the status the image name doesn't seem to change. Not sure why we chose one over the other, probably because of preferring the assurance that the spec won't be modified

Reply:
yes, it should produce the same result by looking at the `V1Container` spec and `V1ContainerStatus` spec. 😄